File: C++/848. Shifting Letters.cpp
class Solution {
public:
    string shiftingLetters(string s, vector<int>& a) {
        int i, p = 0;
        for (i = s.size() - 1; i >= 0; i--) {
            p = (p + a[i]) % 26;
            s[i] = (s[i] - 'a' + p) % 26 + 'a';
        }
        return s;
    }
};

File: JAVA/848. Shifting Letters.java
class Solution {
    public String shiftingLetters(String s, int[] a) {
        int i, p = 0;
        char[] r = s.toCharArray();
        for (i = s.length() - 1; i >= 0; i--) {
            p = (p + a[i]) % 26; 
            r[i] = (char) ((r[i] - 'a' + p) % 26 + 'a'); 
        }
        return new String(r); 
    }
}